src/query/expression/src/kernels/sort.rs

```diff
@@ -239,16 +244,15 @@ fn compare_decimal256(left: &dyn Array, right: &dyn Array) -> ArrowResult<DynCom
 
 fn build_compare(left: &dyn Array, right: &dyn Array) -> ArrowResult<DynComparator> {
     match left.data_type() {
-        ArrowType::Extension(name, _, _) => {
-            if name == "Variant" {
-                compare_variant(left, right)
-            } else {
-                Err(ArrowError::NotYetImplemented(format!(
-                    "Sort not supported for data type {:?}",
-                    left.data_type()
-                )))
-            }
-        }
+        ArrowType::Extension(name, _, _) => match name.as_str() {
+            "variant" => compare_variant(left, right),
+            "EmptyArray" => compare_null(left, right),
+            _ => Err(ArrowError::NotYetImplemented(format!(
+                "Sort not supported for data type {:?}",
+                left.data_type()
+            ))),
+        },
+        ArrowType::Null => compare_null(left, right),
         ArrowType::Decimal256(_, _) => compare_decimal256(left, right),
         _ => arrow_ord::build_compare(left, right),
     }
```

Comment:
Please add some unit tests and SQL logic tests for those cases.
You can refer to the following examples:
https://github.com/datafuselabs/databend/blob/main/src/query/functions/tests/it/scalars/array.rs#L646
https://github.com/datafuselabs/databend/blob/main/tests/sqllogictests/suites/query/02_function/02_0061_function_array#L177

Reply:
> Please add some unit tests and SQL logic tests for those cases. You can refer to the following examples: https://github.com/datafuselabs/databend/blob/main/src/query/functions/tests/it/scalars/array.rs#L646 https://github.com/datafuselabs/databend/blob/main/tests/sqllogictests/suites/query/02_function/02_0061_function_array#L177

I have added the corresponding test case for this change, but I am unsure if it complies with databend's testing specifications. If there is anything incorrect, I would appreciate it if you could point it out. Thank you